README.md

```diff
@@ -73,45 +76,61 @@ var minioClient = new Minio.Client({
   secretKey: 'zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG',
 })
 
-// File that needs to be uploaded.
-var file = '/tmp/photos-europe.tar'
+// Destination bucket
+var bucket = 'js-test-bucket'
 
-// Make a bucket called europetrip.
-minioClient.makeBucket('europetrip', 'us-east-1', function (err) {
-  if (err) return console.log(err)
+// File to upload and name to save as to the bucket
+var source_file = '/tmp/test-file.txt'
+var destination_file = 'my-test-file.txt'
+
+// Create the bucket, if it doesn't already exist
+minioClient.bucketExists(bucket, function (err, exists) {
+  if (err) {
+    return console.log(err)
+  }
+
+  if (exists) {
+      console.log('Bucket ' + bucket + ' exists.')
+  }
+  else {
+      minioClient.makeBucket(bucket, 'us-east-1', function (err) {
+        if (err) return console.log(err)
+
+        console.log('Bucket ' + bucket + ' created successfully in "us-east-1".')
+      })
+}
```

Comment:
How about we start to use modern Javascript syntax?

- Don't use `var`
- Don't `return void` -- in this case, `return console.log()`, but do this instead:
   ```js
   if (err) {
     console.error(err);
     return;
   }
   ```
- Preferably use `async/await` syntax.

Reply:
Good to know this example doesn't reflect current practice, I'll update throughout. For async/await, if you happen to have a JavaScript-relevant link about it that would be super. Otherwise I'm sure Google has an opinion.

I'm from C/C++ with detours through Python and Go, so I can read many languages but am not fluent. The end goal of this update is to have something to start an SDK tutorial in the regular MinIO documentation.

Much appreciate the feedback @aldy505 and @trim21 !

Reply:
For async/await, just create changes for the code, and we'll submit code reviews from there. That should be much faster than forcing you to learn that from scratch.

Reply:
I tried to refresh my reworked example with the changes @trim21 made in `master` recently, including the PR merged today. But I'm not able to run their new version of the sample code. Maybe it wasn't intended to be a full working example?
```
import * as Minio from 'minio'
^^^^^^
SyntaxError: Cannot use import statement outside a module
```
```
await minioClient.makeBucket(bucket, 'us-east-1')
^^^^^
SyntaxError: await is only valid in async functions and the top level bodies of modules
```

Reply:
On your `package.json` you should be adding `"type": "module"`.

The keyword you're looking for is "top level await". See one here https://stackoverflow.com/questions/46515764/how-can-i-use-async-await-at-the-top-level

Reply:
On your `package.json` you should be adding `"type": "module"`.

The keyword you're looking for is "top level await". See one here https://stackoverflow.com/questions/46515764/how-can-i-use-async-await-at-the-top-level

Reply:
I appreciate the link, although I don't know what to do with this information.

If the revised README example (the one currently in `master`) is recommended style, I need to set this update aside until I know a lot more about JavaScript. My attempts to get it to work have only resulted in different errors I don't understand.

Reply:
> I tried to refresh my reworked example with the changes @trim21 made in `master` recently, including the PR merged today. But I'm not able to run their new version of the sample code. Maybe it wasn't intended to be a full working example?
> 
> ```
> import * as Minio from 'minio'
> ^^^^^^
> SyntaxError: Cannot use import statement outside a module
> ```
> 
> ```
> await minioClient.makeBucket(bucket, 'us-east-1')
> ^^^^^
> SyntaxError: await is only valid in async functions and the top level bodies of modules
> ```

create a `example.mjs` file and paste example into it, then you should be able to run it .

Reply:
I know Python and golang， so if you have done updating examples of minio-py and minio-go ， I can help you update examples of minio-js.

Reply:
Ok! As a `.mjs` file this works. I don't yet understand why, but it's progress. Yay.